Assets/_Laz/Tests/PlanetoidTests.cs

```diff
@@ -58,6 +58,96 @@ public void Test_Planetoid_Reset()
             Assert.AreEqual(mockPlanetoidBehaviour.ResetCount, 1, "Reset should be called once");
         }
 
+        [Test]
+        public void Test_Planetoid_WrappableManager_TriggersWin()
+        {
+            //Arrange
+            var numMockPlanets = 5;
+            MockPlanetoidBehaviour[] mockPlanetoids = new MockPlanetoidBehaviour[numMockPlanets];
+            Planetoid[] planetoids = new Planetoid[numMockPlanets];
+            for(int i = 0; i < numMockPlanets; i++)
+            {
+                mockPlanetoids[i] = new MockPlanetoidBehaviour();
+                planetoids[i] = new Planetoid(mockPlanetoids[i]);
+                mockPlanetoids[i].PlanetoidModel = planetoids[i];
+            }
+            var mockStateManager = new MockStateManager();
+            var lazoWrappableManager = new LazoWrappableManager(mockPlanetoids, mockStateManager);
+
+            //Act
+            mockStateManager.SetState(State.Play);
+            for (int i = 0; i < numMockPlanets; i++)
+            {
+                planetoids[i].ActivateLazo();
+            }
+
+            //Assert
+            Assert.AreEqual(mockStateManager.GetState(), State.WinGame, "State should be set to win");
+            for (int i = 0; i < numMockPlanets; i++)
+            {
+                Assert.IsTrue(planetoids[i].IsActivated, $"Planetoid {i} should be activated");
+            }
+        }
+
+        [Test]
+        public void Test_Planetoid_WrappableManager_CleanUp()
+        {
+            // Arrange
+            var numMockPlanets = 5;
+            MockPlanetoidBehaviour[] mockPlanetoids = new MockPlanetoidBehaviour[numMockPlanets];
+            Planetoid[] planetoids = new Planetoid[numMockPlanets];
+            for (int i = 0; i < numMockPlanets; i++)
+            {
+                mockPlanetoids[i] = new MockPlanetoidBehaviour();
+                planetoids[i] = new Planetoid(mockPlanetoids[i]);
+                mockPlanetoids[i].PlanetoidModel = planetoids[i];
+            }
+            var mockStateManager = new MockStateManager();
+            var lazoWrappableManager = new LazoWrappableManager(mockPlanetoids, mockStateManager);
+
+            //Act
+            mockStateManager.SetState(State.Play);
+            for (int i = 0; i < numMockPlanets; i++)
+            {
+                planetoids[i].ActivateLazo();
+            }
+            lazoWrappableManager.CleanUp();
+
+            //Assert
+            Assert.AreEqual(mockStateManager.GetState(), State.WinGame, "State should be set to win");
+            for (int i = 0; i < numMockPlanets; i++)
+            {
+                Assert.IsFalse(planetoids[i].IsActivated, $"Planetoid {i} should not be activated");
```

Comment:
So you're testing `CleanUp`. 
The more accurate thing to test is if `interest.CleanUp` is called. 

So following What you did below with `ResetCount`, you can add a `CleanUpCount`.

Since you don't care if a planetoid is activated or not in this situation, you just care if CleanUp was called. 
And it's not `lazoWrappableManager` responsibility to care what `Planetoid.CleanUp` does inside aka reset `IsActivated`. You just care if the method `interest.CleanUp` is called.

Reply:
This makes sense, just fixed this